graylog2-server/src/main/java/org/graylog2/plugin/lookup/LookupResult.java

```diff
@@ -42,8 +44,18 @@
  */
 @AutoValue
 public abstract class LookupResult {
+    private static final long NO_TTL = Long.MAX_VALUE;
+
+    // Cache erroneous results with a shorter TTL
+    private static final long ERROR_CACHE_TTL = Duration.ofSeconds(5).toMillis();
```

Comment:
I think we should make this TTL user configurable. There will be use cases where you want shorter or longer cache times of errors.

Reply:
Globally? Or per DataAdapter?

Reply:
Per data adapter. Depending on the data you deal with, you might want to use a different TTL for errors.